core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -208,9 +213,22 @@ object ConsoleProducer {
       .defaultsTo(1024*100)
     val propertyOpt = parser.accepts("property", "A mechanism to pass user-defined properties in the form key=value to the message reader. " +
       "This allows custom configuration for a user-defined message reader. Default properties include:\n" +
-      "\tparse.key=true|false\n" +
-      "\tkey.separator=<key.separator>\n" +
-      "\tignore.error=true|false")
+      "\tparse.key=false\n" +
+      "\tparse.headers=false\n" +
+      "\tignore.error=false\n" +
+      "Default parsing pattern when:\n" +
+      "\tparse.headers=true & parse.key=true:\n" +
+      "\t \"h1:v1,h2...\\tkey\\tvalue\"\n" +
+      "\tparse.headers=false & parse.key=true:\n" +
+      "\t \"key\\tvalue\"\n" +
+      "\tparse.headers=true & parse.key=false:\n" +
+      "\t \"h1:v1,h2...\\tvalue\"\n" +
+      "Customize pattern via (defaults shown)\n" +
+      "\tkey.separator=\\t\n" +
+      "\theaders.delimiter=\\t\n" +
+      "\theaders.separator=,\n" +
+      "\theaders.key.separator=:"
+      )
```

Comment:
This is quite hard to read. You could use a multiline string in Scala with `"""   """`. You can take a look at how we did it in the `ConsoleConsumer.`

Also, it would be great if we could have all the parameters on the top followed by the examples.

Reply:
Adjusted